app/api/auth/oauth.py

```diff
@@ -0,0 +1,66 @@
+# app/auth/oauth.py
+
+from datetime import datetime, timezone
+from fastapi import Depends, HTTPException, status
+from fastapi.security import OAuth2PasswordBearer
+from jose import JWTError
+import jwt
+from app.settings import settings
+from app.utils import ALGORITHM
+from app.models import User
+from app.api.auth.auth import fake_db_user
+
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/api/v1/login/")
+
+
+# JWT token verification
+def verify_token(token: str, credentials_exception):
+    try:
+        payload = jwt.decode(token, settings.SECRET_KEY, algorithms=[ALGORITHM])
+        user_id: str = payload.get("sub")
+        exp: int = payload.get("exp")
+        
+        if user_id is None:
+            raise credentials_exception
+        
+        if datetime.fromtimestamp(exp, tz=timezone.utc) < datetime.now(tz=timezone.utc):
+            raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Token has expired")
+
+        return user_id
```

Comment:
the return statement should be outside the try block. Also, the try block should only surround the JWT decoding function.